### PR TITLE
Fix catcher showing miss sprite upon missing bananas

### PR DESCRIPTION
**Pull request by EVAST9919**

### CODE REVIEW

**Comment by bdach** [comment]
Having a test for this behaviour would be nice.

**Comment by EVAST9919** [comment]
We already have one `TestSceneDrawableHitObjects`

**Comment by bdach** [comment]
...which doesn't have asserts, so if this ever breaks the test is useless for checking this logic.

**Comment by EVAST9919** [comment]
Added

File: osu.Game.Rulesets.Catch/UI/CatcherArea.cs [hunk 1]
```diff
@@ -345,21 +345,24 @@ public bool AttemptCatch(CatchHitObject fruit)
                 if (validCatch)
                     updateState(fruit.Kiai ? CatcherAnimationState.Kiai : CatcherAnimationState.Idle);
                 else
-                    updateState(CatcherAnimationState.Fail);
+                {
+                    if (!(fruit is Banana))
+                        updateState(CatcherAnimationState.Fail);
+                }
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/UI/CatcherArea.cs]: can combine else and if here

File: osu.Game.Rulesets.Catch/UI/CatcherArea.cs [hunk 2]
```diff
@@ -345,21 +345,24 @@ public bool AttemptCatch(CatchHitObject fruit)
...
+                if (CurrentState == state)
                     return;
 
-                currentState = state;
+                CurrentState = state;
                 updateCatcher();
             }
 
-            private CatcherAnimationState currentState;
+            public CatcherAnimationState CurrentState;
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/UI/CatcherArea.cs]: private set

**Review by peppy — CHANGES_REQUESTED** [review]

**Pull request merged**